src/graphql/graphql-api.php

```diff
@@ -0,0 +1,255 @@
+		 */
+		$is_graphql_to_be_enabled = apply_filters( 'vip_block_data_api__is_graphql_enabled', true );
+
+		if ( ! $is_graphql_to_be_enabled ) {
+			return;
+		}
+
+		// Register the type corresponding to the attributes of each individual block.
+		register_graphql_object_type(
+			'BlockAttribute',
+			[
+				'description' => 'Block attribute',
+				'fields'      => [
+					'name'  => [
+						'type'        => [ 'non_null' => 'String' ],
+						'description' => 'Block data attribute name',
+					],
+					'value' => [
+						'type'        => [ 'non_null' => 'String' ],
+						'description' => 'Block data attribute value',
+					],
+				],
+			],
+		);
+
+		// Register the type corresponding to the individual block, with the above attribute.
+		register_graphql_type(
+			'BlockData',
+			[
+				'description' => 'Block data',
+				'fields'      => [
+					'id'          => [
+						'type'        => [ 'non_null' => 'ID' ],
+						'description' => 'ID of the block',
+					],
+					'parentId'    => [
+						'type'        => 'ID',
+						'description' => 'ID of the parent for this inner block, if it is an inner block. Otherwise, it will not be set. If it set, this will match the ID of the block',
+					],
+					'name'        => [
+						'type'        => [ 'non_null' => 'String' ],
+						'description' => 'Block name',
+					],
+					'attributes'  => [
+						'type'        => [
+							'list_of' => 'BlockAttribute',
+						],
+						'description' => 'Block attributes',
+					],
+					'innerBlocks' => [
+						'type'        => [ 'list_of' => 'BlockData' ],
+						'description' => 'Flattened list of inner blocks of this block',
+					],
+				],
+			],
+		);
+
+		// Register the type corresponding to the list of individual blocks, with each item being the above type.
+		register_graphql_type(
+			'BlocksData',
+			[
+				'description' => 'Data for all the blocks',
+				'fields'      => [
+					'blocks'   => [
+						'type'        => [ 'list_of' => 'BlockData' ],
+						'description' => 'List of blocks data',
+						'args'        => [
+							'flatten' => [
+								'type'        => 'Boolean',
+								'description' => 'Collate the inner blocks under each root block into a single list with a parent-child relationship. This is set to true by default, and setting it to false will reverse that to preserve the original block hierarchy, at the expense of knowing the exact depth when querying the inner blocks. Default: true',
+							],
+						],
+						'resolve'     => function ( $blocks, $args ) {
+							if ( ! isset( $args['flatten'] ) || true === $args['unflatten'] ) {
+								$blocks['blocks'] = array_map(function ( $block ) {
+									// Flatten the inner blocks, if any.
+									if ( isset( $block['innerBlocks'] ) ) {
+										$block['innerBlocks'] = self::flatten_inner_blocks( $block['innerBlocks'], $block['id'] );
+									}
+
+									return $block;
+								}, $blocks['blocks'] );
```

Comment:
You can just do this, no?
```suggestion
								// Flatten the inner blocks, if any.
								$blocks['blocks'] = self::flatten_inner_blocks( $blocks['blocks'], null );
```

Reply:
This replacement doesn't work as-is, because `flatten_inner_blocks()` needs the `$parent_id` parameter, which is passed to `innerBlocks` to determine their parent IDs:

https://github.com/Automattic/vip-block-data-api/blob/d578fe470a0ae1248eb0198b682ef76d886bf3ff/src/graphql/graphql-api.php#L116-L119

I think this is due to the design that top level blocks are treated differently as "root" blocks, but their children are flattened beneath them all together. I'm sure it's possible to rewrite `flatten_inner_blocks()` so this is done internally, but I think this works fine.

Reply:
Wait a minute, I'm rereading this and I'm not sure why it's not working. Let me take a second look.